webpack/assets/javascripts/react_app/components/ReportTemplate/ReportTemplateInput.js

```diff
@@ -0,0 +1,70 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import DateTime from './Inputs/ReportDateTime';
+import Autocomplete from './Inputs/ReportAutoComplete';
+
+const TYPES = { PLAIN: 'plain', DATE: 'date', SEARCH: 'search' };
```

Comment:
consider adding those TYPES in a `ReportTemplateConstants.js` file instead for external use,
and than you could import them destructively and make the switch cases cleaner.

Reply:
is it possible to somehow connect this with the same definition on backend?